docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+-  Responds to CLI commands.
+-  Hosts the WebUI, which is primarily used to monitor experiments.
+-  Serves the REST API.
+
+The master is also responsible for the following training-specific searcher, metrics, checkpointing,
+and scheduling functionality:
+
++---------------+----------------------------------------------------------------------+
+| Function      | Description                                                          |
++===============+======================================================================+
+| Searcher      | The *Searcher* implements a hyperparameter search algorithm and is   |
+|               | responsible for coordinating the work of all experiment trials.      |
++---------------+----------------------------------------------------------------------+
+| Metrics       | *Metrics* provides persistent storage of metrics reported by Trials. |
++---------------+----------------------------------------------------------------------+
+| Checkpointing | *Checkpointing* captures the training state at a given time in the   |
+|               | *model registry*.                                                    |
++---------------+----------------------------------------------------------------------+
+| Scheduling    | *Scheduling* schedules jobs to run, ensuring that all of the compute |
+|               | resources required for a job are available before the job launches.  |
++---------------+----------------------------------------------------------------------+
+
+Agent Node Scope of Responsibility
+----------------------------------
+
+Agent node responsibilities:
+
+-  Discovers local computing devices and sends device/slot metadata to the master.
+-  Runs workloads at the request of the master.
+-  Monitors containers and sends container information to the master.
+-  For a *trial runner*, which runs a trial in a containerized environment, reports trial runner
+   states to the master.
+
+The agent manages a number of GPU or CPU devices, which are referred to as *slots*. There is
+typically one agent per compute server and the active experiment volume dictates the number of
+agents needed.
+
+Agents communicate only with the master and state information is not maintained on the agent.
+
+PostgreSQL Database
+===================
+
+Each cluster requires access to a `PostgreSQL <https://www.postgresql.org/>`_ database to store
+experiment and trial metadata. Although not required, the database typically resides on the master.
+If you use the ``det deploy`` command to launch Determined on a cloud provider or on-premises,
+PostgreSQL preinstalls a network file system shared by all agents. Otherwise, you need to manually
+install PostgreSQL.
+
+Docker Images
+=============
+
+Determined launches workloads using `Docker <https://www.docker.com/>`_ containers. By default,
+workloads execute inside a Determined-provided container that includes common deep learning
+libraries and frameworks.
```

Comment:
> By default, workloads execute inside a Determined-provided container that includes common deep learning libraries and frameworks.

At first glance, I thought the "default" was running in a container, but you didn't have to.  Suggested rephrase:

> Determined provides a default container that includes common deep learning libraries and frameworks.